chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/impl/TestingFunctionalityManagerImpl.java

```diff
@@ -1,84 +1,108 @@
 package gov.healthit.chpl.manager.impl;
 
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.ContextRefreshedEvent;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
 import gov.healthit.chpl.dao.TestFunctionalityDAO;
 import gov.healthit.chpl.domain.TestFunctionality;
+import gov.healthit.chpl.dto.TestFunctionalityCriteriaMapDTO;
 import gov.healthit.chpl.dto.TestFunctionalityDTO;
 import gov.healthit.chpl.manager.TestingFunctionalityManager;
 
 @Service
-public class TestingFunctionalityManagerImpl implements TestingFunctionalityManager {
+@Transactional
+public class TestingFunctionalityManagerImpl implements TestingFunctionalityManager, ApplicationListener<ContextRefreshedEvent> {
 
-    @Autowired
     private TestFunctionalityDAO testFunctionalityDAO;
+    
+    private Map<String, List<TestFunctionalityDTO>> testFunctionalityByCriteria2015 = 
+            new HashMap<String, List<TestFunctionalityDTO>>();
+
+    private Map<String, List<TestFunctionalityDTO>> testFunctionalityByCriteria2014 = 
+            new HashMap<String, List<TestFunctionalityDTO>>();
+    
+    @Autowired
+    public TestingFunctionalityManagerImpl(TestFunctionalityDAO testFunctionalityDAO) {
+        this.testFunctionalityDAO = testFunctionalityDAO;
+    }
+    
+    @Override
+    public void onApplicationEvent(ContextRefreshedEvent event) {
```

Comment:
When does this get called?

Reply:
I wanted to run this code in a @PostConstruct, but not all of the beans that I needs to use in the logic were initialized.  This is an event that is raised by the spring context when all of the beans have been initialized.  It does not need to be explicitly called. 
But in some of the unit tests that are not dependent on the spring context, it does need to be called explicitly.

Reply:
Okay... I guess I'm asking because we haven't done this anywhere else in the code. I have seen PostConstruct used and don't understand why it doesn't work here... which beans are not initialized? Also have seen in CertifiedProductDetailsManagerImpl in the Autowired constructor after each member of the class is set some data is loaded (loadCqms and loadCriteriaMacraMeasures are called). Could that work here instead? It doesn't seem right to have to call this from the unit tests or anywhere.

Reply:
For some reason, when I wrote the code the the DAO was not initialized when the @PostConstruct ran.  I'm going to give this a try:  https://nixmash.com/post/using-postconstruct-and-dependson-in-spring